sdk/dotnet/Pulumi.Tests/Core/InputTests.cs

```diff
@@ -44,6 +45,22 @@ public Task MergeInputMaps()
                 Assert.True(map2Data.Value.ContainsValue("V3"));
             });
 
+        [Fact]
+        public Task InputMapCollectionInitializers()
+            => RunInPreview(async () =>
+            {
+                var map = new InputMap<string>
+                {
+                    { "K1", "V1" },
+                    { "K2", Output.Create("V2") },
+                    new Dictionary<string, string> { { "K3", "V3" }, { "K4", "V4"} },
```

Comment:
I find it slightly strange that we're allowing the adding of a collection here? None of the standard dotnet collections support this use of collection initializers afaik.

Reply:
Yes, this is more about making the collection initializers more user friendly than about following other dotnet conventions. These collections are used extensively by resources so it helps a lot with inline resource instantiation and therefore simplifies dotnet code using pulumi libraries.

Reply:
Do you have any examples of real world uses cases using this. We're not against adding a non-standard Add method but we'd like to see that it is genuinely useful to have. If there's anything in pulumi/examples, or a new example we could add there to show off the wins of having these methods that would be really good.

Reply:
Yes, mainly with kubernetes resources with multiple dependencies, e.g.:
```c#
var certManagerRelease = new Release("cert-manager",
    new ReleaseArgs
    {
        Namespace = "cert-manager",
        Name = "cert-manager",
        RepositoryOpts = new RepositoryOptsArgs { Repo = "https://charts.jetstack.io" },
        Chart = "cert-manager",
        Version = K8sConfig.CertManagerChartVersion,
        Values = certManagerValues,
        Atomic = true,
        SkipCrds = true
    },
    new CustomResourceOptions { DependsOn = { certManagerCrds.Ready(), certManagerNs, kubePrometheusStackCrds.Ready() }, Provider = k8sProvider });
```
This does not compile without my proposed changes.